src/test/java/BaseTest.java

```diff
@@ -1,18 +1,78 @@
 import io.github.bonigarcia.wdm.WebDriverManager;
+import org.openqa.selenium.By;
 import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.WebElement;
 import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.chrome.ChromeOptions;
 import org.openqa.selenium.firefox.FirefoxDriver;
+import org.openqa.selenium.interactions.Actions;
 import org.openqa.selenium.safari.SafariDriver;
-import org.testng.annotations.BeforeMethod;
-import org.testng.annotations.BeforeSuite;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.annotations.*;
+
+import java.time.Duration;
 
 
 public class BaseTest {
 
+    public static WebDriver driver = null;
+    public static WebDriverWait wait;
+
+    public static Actions actions = null;
+    public static By emailField = By.cssSelector("[type='email']");
+    public static By passwordField = By.cssSelector("[type='password']");
+    public static By submitButton = By.cssSelector("[type='submit']");
+
+
+
     @BeforeSuite
     static void setupClass() {
         WebDriverManager.chromedriver().setup();
     }
 
-}
+    @BeforeMethod
+    @Parameters({"BaseURL"})
+    public static void launchBrowser(String BaseURL) {
+        driver = new ChromeDriver();
+        actions = new Actions(driver);
+        wait = new WebDriverWait(driver, Duration.ofSeconds(5));
+        //        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(5));
+        driver.get(BaseURL);
+        driver.manage().window().maximize();
+    }
+
+//    @AfterClass
```

Comment:
@AfterMethod annotated method will be run after each test method i.e say there are three test methods (i.e test cases), then @AfterMethod annotated method will be called thrice after each test method.

```
  @AfterMethod
  public static void closeBrowser(){
      driver.quit();
   }
```